@attr(tags=['advanced'], required_hardware='false')
def test_deployVM_in_sharedNetwork_as_domainadmin_scope_all_crossdomainuser(self):
    '\n\tValiate that Domain admin is NOT able to deploy a VM for user in other domain in a shared network with scope=all\n\t'
    self.apiclient.connection.apiKey = self.user_d1_apikey
    self.apiclient.connection.securityKey = self.user_d1_secretkey
    self.vmdata['name'] = (self.acldata['vmROOTA']['name'] + '-shared-scope-all')
    self.vmdata['displayname'] = (self.acldata['vmROOTA']['displayname'] + '-shared-scope-all')
    try:
        vm = VirtualMachine.create(self.apiclient, self.vmdata, zoneid=self.zone.id, serviceofferingid=self.service_offering.id, templateid=self.template.id, networkids=self.shared_network_all.id, accountid=self.account_d2a.name, domainid=self.account_d2a.domainid)
        self.fail('Domain admin user is able to Deploy VM for a domain user he does not have access to in a shared network with scope=domain with no subdomain access ')
    except Exception as e:
        self.debug(('When a Domain admin user  deploys a VM for a domain user he does not have access to in a shared network with scope=domain with no subdomain access %s' % e))
        if (not CloudstackAclException.verifyMsginException(e, CloudstackAclException.NO_PERMISSION_TO_OPERATE_DOMAIN)):
            self.fail('Error mesage validation failed when Domain admin user tries to Deploy VM for a domain user he does not have access to in a shared network with scope=domain with no subdomain access ')
